engine/access/rest/websockets/controller.go

```diff
@@ -373,7 +404,24 @@ func (c *Controller) handleMessage(ctx context.Context, message json.RawMessage)
 	return nil
 }
 
+// handleSubscribe processes a subscription request.
+//
+// Expected error returns during normal operations:
+//   - ErrMaxSubscriptionsReached: if the maximum number of active subscriptions per connection is exceeded.
+//   - context.Canceled: if the operation is canceled, during an unsubscribe action.
 func (c *Controller) handleSubscribe(ctx context.Context, msg models.SubscribeMessageRequest) {
+	// Check if the maximum number of active subscriptions per connection has been reached.
+	// If the limit is exceeded, an error is returned, and the subscription request is rejected.
+	if uint64(c.dataProviders.Size()) > c.config.MaxSubscriptionsPerConnection {
```

Comment:
we should allow up to and including `c.config.MaxSubscriptionsPerConnection` not more
```suggestion
	if uint64(c.dataProviders.Size()) >= c.config.MaxSubscriptionsPerConnection {
```